src/components/StyleCard/StyleCard.tsx

```diff
@@ -61,74 +71,131 @@ export const StyleCard = ({
+
   return (
     <div
       className={styling.styleCard}
       key={index}
-      data-status={preview.isValid ? "valid" : "invalid"}>
+      data-status={isError ? "invalid" : "valid"}
+    >
       <Button
-        type='button'
+        type="button"
         leftIcon={<DeleteIcon />}
-        buttonType='negative'
+        buttonType="negative"
         style={{ backgroundColor: "transparent", color: "var(--alert-color)" }}
         className={styling.deleteButton}
-        onClick={() => handleStyleRemoval(index)}></Button>
+        onClick={() => handleStyleRemoval(index)}
+      ></Button>
       <Section
         headerClassName={styling.header}
         title={<StyleTitle {...style} />}
         collapsible={{
           isOpen: style.isOpen,
           onToggle: () =>
-            handleStyleChange(index, "isOpen", !style.isOpen),
-        }}>
+            handleStyleChange(index, "isOpen", !style.isOpen as T[keyof T]),
+        }}
+      >
         <FieldGroup key={index} className={styling.content}>
           <TextField
             id={`slug-${index}`}
             required
-            name='slug'
-            label='Slug (to be used as a CSS class)'
+            name="slug"
+            label="Slug (to be used as a CSS class)"
             value={style.slug}
             onChange={(newValue) =>
-              handleStyleChange(index, "slug", newValue)
+              handleStyleChange(index, "slug", newValue as T[keyof T])
             }
             error={slugValidation.error}
           />
           <TextField
             id={`title-${index}`}
             required
-            name='title'
-            label='Title (shown in the Structured Text editor)'
+            name="title"
+            label="Title (shown in the Structured Text editor)"
             value={style.title}
             onChange={(newValue) =>
-              handleStyleChange(index, "title", newValue)
+              handleStyleChange(index, "title", newValue as T[keyof T])
             }
             error={titleValidation.error}
           />
-          <SelectField
-            id={`nodes-${style.slug}-${index}`}
-            name='nodes'
-            label='Nodes'
-            value={style.nodes}
-            selectInputProps={{
-              isMulti: true,
-              options: NODE_OPTIONS,
+          {"nodes" in style && (
+            <SelectField
+              id={`nodes-${style.slug}-${index}`}
+              name="nodes"
+              label="Nodes"
+              value={style.nodes}
+              selectInputProps={{
+                isMulti: true,
+                options: NODE_OPTIONS,
+              }}
+              onChange={(newValue) =>
+                handleStyleChange(
+                  index,
+                  "nodes" as keyof T,
+                  newValue as T[keyof T],
+                )
+              }
+            />
+          )}
+          {"icon" in style && (
+            <TextField
```

Comment:
Consider making this a select field, with perhaps the icons in front of the value (https://react-select.com/home#custom-styles or https://react-select.com/components#replacing-components)